homeassistant/components/growatt/sensor.py

```diff
@@ -0,0 +1,178 @@
+"""
+Support for Growatt Plant energy production sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.growatt/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ["growatt==0.0.2"]
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt Plant sensor."""
+    import growatt
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    growatt_client = growatt.GrowattApi()
+
+    is_login_success = login(growatt_client, username, password)
+    if is_login_success:
+        sensor_today = GrowattPlantToday(
+            hass, growatt_client, username, password
+        )
+        sensor_total = GrowattPlantTotal(
+            hass, growatt_client, username, password
+        )
+        sensor_current = GrowattPlantCurrent(
+            hass, growatt_client, username, password
+        )
+        add_entities([sensor_today, sensor_total, sensor_current])
+        return True
```

Comment:
no one checks setup_platform return values, might as well remove them

Reply:
👍 didn't know that, good to know!